File: RouterLib/src/main/java/com/kronos/router/model/HostParams.java
package com.kronos.router.model;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by zhangyang on 16/7/16.
 */
public class HostParams {
    private String host;

    public HostParams(String host) {
        this.host = host;
    }

    private final Map<String, RouterOptions> _routes = new HashMap<String, RouterOptions>();

    public void setRoute(String path, RouterOptions options) {
        _routes.put(path, options);
    }

    public RouterOptions getOptions(String path) {
        return _routes.get(path);
    }

    public Map<String, RouterOptions> getRoutes() {
        return _routes;
    }

    public String getHost() {
        return host;
    }
}

File: RouterLib/src/main/java/com/kronos/router/model/HostParams.kt
package com.kronos.router.model

import java.util.HashMap

/**
 * Created by zhangyang on 16/7/16.
 */
class HostParams(val host: String) {
    val routes = HashMap<String, RouterOptions>()

    fun setRoute(path: String, options: RouterOptions) {
        if (routes.containsKey(path)) {
            val oldOptions = routes[path]
            if (oldOptions?.weight ?: 0 < options.weight) {
                routes[path] = options
            }
        } else {
            routes[path] = options
        }
    }

    fun getOptions(path: String): RouterOptions? {
        return routes[path]
    }
}
